Document HTTP Client Errors (enableCaptureFailedRequests) (#6987)
getsentry/sentry-docs: src/platforms/dotnet/common/configuration/http-client-errors.mdx

---
title: HTTP Client Errors
sidebar_order: 25
description: "This feature, once enabled, automatically captures HTTP client errors, like bad response codes, as error events and reports them to Sentry"
---

> Once enabled, this feature automatically captures HTTP client errors, like bad response codes, as error events and reports them to Sentry. The error event will contain the `request` and `response` data, such as `url`, `status_code`, and so on.

Reviewer: ```suggestion
Once enabled, this feature automatically captures HTTP client errors, like bad response codes, as error events and reports them to Sentry. The error event will contain information from the HTTP request and response, such as `url`, `status_code`, and so on.
```

To enable it:

```csharp
// Add this to the SDK initialization callback
options.CaptureFailedRequests = true;
```

```fsharp
// Add this to the SDK initialization callback
options.CaptureFailedRequests <- true
```

HTTP client errors will then be captured for any requests from the Sentry SDK. To capture HTTP client errors for outbound requests from your own application, pass `SentryHttpMessageHandler` as the inner handler when creating your `HttpClient`:

```csharp
using var httpClient = new HttpClient(new SentryHttpMessageHandler());
```

```fsharp
let httpClient = new HttpClient(new SentryHttpMessageHandler())
```

> 

Reviewer: ```suggestion

<Alert title="Tip">

You can skip the above step when using any of the Sentry .NET SDKs that are registered via dependency injection (`Senry.AspNetCore`, Sentry.Maui`, `Sentry.Extensions.Logging`, etc.).

Instead, create your `HttpClient` using the `IHttpClientFactory` pattern, as described in [the documentation from Microsoft](https://learn.microsoft.com/aspnet/core/fundamentals/http-requests).  The `SentryHttpMessageHandler` will automatically be added to the factory during initialization, so no additional wire-up is needed.

</Alert>

```

> By default, only HTTP client errors with a response code between `500` and `599` are captured as error events, but you can change this behavior by setting the `FailedRequestStatusCodes` option:

Reviewer: ```suggestion
By default, only HTTP client errors with a response code between `500` and `599` are captured as error events, but you can change this behavior by modifying the `FailedRequestStatusCodes` option:
```

```csharp

> options.FailedRequestStatusCodes = new []{ new HttpStatusCodeRange(400, 599) };

Reviewer: Though replacing the list is valid, I think it would less confusing in docs to show adding to it.
```csharp
options.FailedRequestStatusCodes.Add((400, 499));
```
